types/eslint/index.d.ts

```diff
@@ -0,0 +1,386 @@
+// Type definitions for eslint 4.16
+// Project: https://eslint.org
+// Definitions by: Pierre-Marie Dartus <https://github.com/pmdartus>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface Ast {
+    comments: any[],
+    tokens: any[],
+    loc: any,
+    range: any[];
+}
+
+export type AstNode = any;
+export type Token = any;
+export type Comment = any;
```

Comment:
Can you import these from [`estree`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/estree/index.d.ts)? That should allow for better type checking.